src/library/scala/collection/IndexedSeqView.scala

```diff
@@ -96,23 +96,20 @@ object IndexedSeqView {
         r
       } else Iterator.empty.next()
 
-    override def drop(n: Int): Iterator[A] = {
-      if (n > 0) {
-        pos -= n
-        remainder = Math.max(0, remainder - n)
+    override protected def sliceIterator(from: Int, until: Int): Iterator[A] = {
+      if (_hasNext) {
+        if (remainder <= from) remainder = 0
+        else if (from <= 0) {
+          if (until >= 0 && until < remainder) remainder = until
```

Comment:
I'm not sure this `if` is correct. what about `sliceIterator(1, -2)` with `remainder = <arbitrary>`—shouldn't that set `remainder = 0`? or perhaps throw an `IllegalArgumentException`?

Reply:
The doc on sliceIterator is that until < 0 means no limit. That is drop semantics.

Reply:
I'll reread and re-evaluate my comments based on that